### PR TITLE
[MAINT] run test and flake8 github actions

### LINKED ISSUE #125: run tests on github action instead of travis
- should probably also take care of sending the coverage report to codecov

### LINKED ISSUE #129: fix broken badge in readme

### PR DESCRIPTION
**Pull request by Remi-Gau**
- [x] upload coverage report
- [x] make sure tests pass
- [x] rm travis.yml
- [x] add GHA workflow
- [x] add test data instead of downloading from neurovault to minimize failures
- [x] add pre-commit to run flake8 locally upon commit

---

- closes #125, #126
- closes #129

### CODE REVIEW

File: atlasreader/atlasreader.py [hunk 1]
```diff
@@ -256,7 +256,7 @@ def get_subpeak_coords(clust_img, min_distance=20):
     data = check_niimg(clust_img).get_fdata()
 
     # find local maxima, excluding peaks that are on the border of the cluster
-    local_max = peak_local_max(data, exclude_border=1, indices=False)
+    local_max = peak_local_max(data, exclude_border=1)
```
Comment (Remi-Gau) [inline atlasreader/atlasreader.py:259]: this was dropped in scikit-image 0.20 but removing it leads to a crash

**Comment by Remi-Gau** [comment]
not sure why we are getting this: https://github.com/miykael/atlasreader/actions/runs/6030134696

```
The runner has received a shutdown signal
```

File: atlasreader/tests/test_atlasreader.py [hunk 2]
```diff
@@ -64,12 +67,15 @@ def test_get_atlases():
     with pytest.raises(ValueError):
         atlasreader.get_atlas('not_an_atlas')
 
-
-def test_check_atlases():
+def test_check_atlases_all():
```
Comment (Remi-Gau) [inline atlasreader/tests/test_atlasreader.py:70]: when executing all of those at once it would crash the workflow

**Comment by Remi-Gau** [comment]
Will let this rest for today. Finishing fixing this may require diving into libraries I am not familiar with.

**Comment by miykael** [comment]
I was able to make the tests pass, and updated the faulty badges. But now it crashes do to some coveralls issue. Unfortunately I'm not familiar with coveralls or its github-actions. I registered on coveralls.io and added the a token via the command line, but it's still failing. Any idea how to debug this?

**Comment by Remi-Gau** [comment]
> Any idea how to debug this?

the action was fine I just had forgotten to generate the coverage report. 🙈 


BTW: added python 3.12 in CI

**Comment by miykael** [comment]
Oh super! Thank you for all of this! Looks good to me, will go ahead and merge it!

**Pull request merged**